.upstream-tests/test/std/atomics/atomics.types.operations/atomics.types.operations.wait/atomic_ref_member_wait.pass.cpp

```diff
@@ -0,0 +1,69 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// UNSUPPORTED: libcpp-has-no-threads
+// UNSUPPORTED: c++98, c++03
+// UNSUPPORTED: pre-sm-70
+
+// <cuda/std/atomic>
+
+#include <cuda/std/atomic>
+#include <cuda/std/type_traits>
+#include <cuda/std/cassert>
+
+#include "test_macros.h"
+#include "../atomics.types.operations.req/atomic_helpers.h"
+#include "concurrent_agents.h"
+#include "cuda_space_selector.h"
+
+template <class T, template<typename, typename> typename Selector, cuda::thread_scope Scope>
+struct TestFn {
+  __host__ __device__
+  void operator()() const {
+    typedef cuda::std::atomic_ref<T> A;
+
+#ifdef __CUDA_ARCH__
+    __shared__
+#endif
+    T * t;
+#ifdef __CUDA_ARCH__
+    if (threadIdx.x == 0) {
```

Comment:
This does not really work because it lets a get out of scope

Reply:
Fixed with latest commit:

https://github.com/NVIDIA/libcudacxx/pull/379/commits/63781797164da3ba27bc10faf33c40c83aee51cc